src/Domain/LeanCode.DomainModels/Ids/ITypedId.cs

```diff
@@ -0,0 +1,36 @@
+using System.ComponentModel;
+using System.Diagnostics.CodeAnalysis;
+
+namespace LeanCode.DomainModels.Ids;
+
+[SuppressMessage("?", "CA1000", Justification = "Roslyn bug.")]
+[EditorBrowsable(EditorBrowsableState.Never)]
+public interface IPrefixedTypedId<TSelf> : IEquatable<TSelf>, IComparable<TSelf>
+    where TSelf : struct, IPrefixedTypedId<TSelf>
+{
+    string Value { get; }
+    public static abstract int RawLength { get; }
+    public static abstract TSelf Parse(string? v);
+
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public static abstract System.Linq.Expressions.Expression<Func<string, TSelf>> FromDatabase { get; }
```

Comment:
Nit, but `using System.Linq.Expressions;` at the top maybe?

Reply:
+1